src/coreclr/jit/gentree.cpp

```diff
@@ -11702,6 +11702,66 @@ GenTree* Compiler::gtFoldExprCall(GenTreeCall* call)
     return call;
 }
 
+#ifdef FEATURE_HW_INTRINSICS
+
+//------------------------------------------------------------------------
+// gtFoldHWIntrinsic: see if a HWIntrinsic is foldable
+//
+// Arguments:
+//    node - HWIntrinsic node to examine
+//
+// Returns:
+//    The original node if no folding happened.
+//    An alternative tree if folding happens.
+//
+// Notes:
+//    Checks for HWIntrinsic nodes to Vector64.Create/Vector128.Create/Vector256.Create,
+//    and if the call is to one of these, attempt to optimize.
+
+GenTree* Compiler::gtFoldHWIntrinsic(GenTreeHWIntrinsic* node)
+{
+    // Defer folding if not optimizing.
+    if (opts.OptimizationDisabled())
+    {
+        return node;
+    }
+
+    switch (node->GetHWIntrinsicId())
+    {
+        case NI_Vector128_Create:
+#if defined(TARGET_XARCH)
+        case NI_Vector256_Create:
+#elif defined(TARGET_ARM64)
+        case NI_Vector64_Create:
+#endif
+        {
+            bool hwAllArgsAreConstZero = true;
+            for (GenTree* arg : node->Operands())
+            {
+                if (!arg->IsIntegralConst(0) && !arg->IsFloatPositiveZero())
```

Comment:
Its worth noting that this covers a single case of `0`. There are also two other cases likely worth covering:
1. All bits set
2. All constants are the same

For the first case, we have a special `NI_VectorXXX_get_AllBitsSet` intrinsic and it should be picked when all constants are `-1` for signed or "max value" for unsigned.

For the second case, it allows us to have a much simpler tree. That is, a user may do `Vector128.Create(5, 5, 5, 5)` and we could simplify it down to just `Vector128.Create(5)`

Reply:
-- Just noting, I don't think they have to be handled in this PR

But, it'd be good if we have a tracking issue to cover them otherwise as I expect its fairly easy to extend the checks here.